### PR TITLE
added new command to create new SQL query document

### PR DESCRIPTION
**Pull request by llali**
- Added new command "mssql: New query" TODO: not sure what to use for the short but
- The new command creates new SQL query untitled document in the temp folder but because it's untitled, the file doesn't get created so user can decide to save it or not

- the new query command activates the extension is not already activated. 
- The name of the document will be SQLQuery[number].sql 
- after a new SQL query is created the new connection command will be called for user to open a connection

### CODE REVIEW

File: src/controllers/mainController.ts [hunk 1]
```diff
@@ -52,7 +54,10 @@ export default class MainController implements vscode.Disposable {
         }
         if (vscodeWrapper) {
             this._vscodeWrapper = vscodeWrapper;
```
Comment (benrr101) [inline src/controllers/mainController.ts]: We can simplify this with the or operator into `this._vscodeWrapper = vscodeWrapper || new VscodeWrapper()`
Or we could use a ternary operator. But I think we should avoid using if/else for this kind of logic

File: src/controllers/untitledSqlDocumentService.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import VscodeWrapper from './vscodeWrapper';
+import vscode = require('vscode');
+import path = require('path');
+import os = require('os');
+const fs = require('fs');
+
+/**
+ * Service for creating untitled documents for SQL query
+ */
+export default class UntitledSqlDocumentService {
```
Comment (benrr101) [inline src/controllers/untitledSqlDocumentService.ts]: I'm sure you explored this already, but I feel like there has to be some command built into VS Code to just create an untitled document. I'm thinking by opening an in-memory untitled doc, the header for the file would be "untitled 1" or some such. Can we verify that there isn't any way to programmatically create an untitled document before we go ahead with this change?

File: test/untitledSqlDocumentService.test.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import * as TypeMoq from 'typemoq';
```
Comment (benrr101) [inline test/untitledSqlDocumentService.test.ts]: I see there's a good number of tests added here, but Coveralls says this change drops code coverage by 32%. That's a huge change. Can we make sure we understand what's going on with coveralls's assessment before we move forward?

File: src/controllers/untitledSqlDocumentService.ts (hunk 2 above)
Comment (llali) [inline src/controllers/untitledSqlDocumentService.ts]: that one doesn't let me change the language to sql

---
In reply to: [94176030](https://github.com/Microsoft/vscode-mssql/pull/569#discussion_r94176030) [](ancestors = 94176030)

File: test/untitledSqlDocumentService.test.ts (hunk 3 above)
Comment (llali) [inline test/untitledSqlDocumentService.test.ts]: yes not sure why. I see a permission error in travis job, might be related 

---
In reply to: [94176306](https://github.com/Microsoft/vscode-mssql/pull/569#discussion_r94176306) [](ancestors = 94176306)

File: src/controllers/untitledSqlDocumentService.ts (hunk 2 above)
Comment (benrr101) [inline src/controllers/untitledSqlDocumentService.ts]: Is it possible to change the language of the document after it's created? I took a look at the API but it claims that the languageId of a TextDocument is readonly. I can't really understand why that would be the case, and I'd want further clarification from VS Code folks.

Comment (llali) [inline src/controllers/untitledSqlDocumentService.ts]: I tried that also and it didn't work. I created a feature request in VsCode: https://github.com/Microsoft/vscode/issues/17917#issuecomment-269953344

Comment (benrr101) [inline src/controllers/untitledSqlDocumentService.ts]: Fair enough, seems silly they don't have a programmatic way of doing that, but it's a project that's still growing.

**Pull request merged**